build.sbt

```diff
@@ -152,10 +152,34 @@ lazy val amm = project
 
 
     assemblyOption in assembly := (assemblyOption in assembly).value.copy(
-      prependShellScript = Some(
+      prependShellScript = {
+        def universalScript(shellCommands: Seq[String],
+                            cmdCommands: Seq[String],
+                            shebang: Boolean = true): Seq[String] = {
+          Seq(
+            Seq("#!/usr/bin/env sh")
+              .filter(_ => shebang),
+            Seq("shopt -s expand_aliases", "alias ::=''")
+              .map(line => s":; $line"),
+            (shellCommands :+ "exit")
+              .map(line => s":: $line"),
+            "@echo off" +: cmdCommands :+ "exit /B",
+            Seq("\r\n")
+          ).flatten
+        }
+
+        def defaultUniversalScript(javaOpts: Seq[String] = Seq.empty, shebang: Boolean = true): Seq[String] = {
+          val javaOptsString = javaOpts.map(_ + " ").mkString
+          universalScript(
+            shellCommands = Seq(s"exec java -jar $javaOptsString" + """$JAVA_OPTS "$0" "$@""""),
+            cmdCommands = Seq(s"java -jar $javaOptsString" + """%JAVA_OPTS% "%~dpnx0" %*"""),
```

Comment:
Though it's not necessary at this point as the commands are single-line now, it would be nice to use multiline `String`s here instead of `Seq[String]` and have `shellCommands` and `cmdCommands` split by `universalScript` just in case the scripts need to be adjusted in the future to include more lines.

Reply:
I'm not quite sure if I understand this correctly but multiple lines would be possible like this:
```Seq("line1", "line2")```
or like that:
```Seq("line1\nline2")```

Reply:
Yes, but one would rather write:

```scala
"""line1
  |line2""".stripMargin
```

Though, I think using what you have and passing:

```scala
"""line1
  |line2""".stripMargin.split('\n')
```

would work too. So, nevermind.